src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -72,6 +73,9 @@ private void checkCommitTimeout() {
 
     private void shutdownGracefully(final String reason) {
         getLog().info("Shutting down gracefully. Reason: {}", reason);
+        offsets.entrySet().stream().findFirst()
+                .ifPresent(pk -> flushData(pk.getKey(), new TreeMap<>(), Optional.of(reason)));
```

Comment:
Yes. I though about changing the API in order to send a batch with only metadata. Wdyt?